fix(migrations): avoid generating imports with forward slashes

crisbeto opened the pull request:
We're using Node's `path` utilities to remap existing imports and create new ones which can yield paths with forward slash separators.

These changes add some logic to ensure that we only generate forward slashes.

alan-agius4 commented on packages/core/schematics/ng-generate/standalone-migration/util.ts:
```diff
@@ -257,5 +257,6 @@ export function getRelativeImportPath(fromFile: string, toFile: string): string
 
-  return path;
+  // Using the Node utilities can yield paths with forward slashes on Windows.
+  return path.replace(/\\/g, '/');
```
FYI Given that both paths have been normalized before (Which is the case for TS and Devkit), you can use the `posix` path methods to avoid these. 

I would find it confused (and maybe others as well) that non normalised paths are are used through out. Typically in devkit and schematics paths are always posix.

crisbeto replied:
I'm not sure if the TS compiler would produce posix when it creates the `SourceFile`s though.

alan-agius4 replied:
TS always uses posix paths.

devversion commented on packages/core/schematics/utils/import_manager.ts:
```diff
@@ -67,6 +67,11 @@ export class ImportManager {
+    // can yield a path with back slashes. Normalize them since outputting such
+    // paths will also cause TS to escape the forward slashes.
+    moduleName = moduleName.replace(/\\/g, '/');
+
```
I like the normalization above, just to be safe, even if TS uses posix.

I think the import manager should use the module name as is though. If you don't fee strongly, would prefer not doing that since we don't deal with paths necessarily, and also don't relativize paths etc.

crisbeto replied:
As discussed over DM, I'll move this into the standalone-specific `ChangeTracker`.

devversion reviewed (approved):
lgtm aside from my comment, if possible

crisbeto committed:
fix(migrations): avoid generating imports with forward slashes

We're using Node's `path` utilities to remap existing imports and create new ones which can yield paths with forward slash separators.

These changes add some logic to ensure that we only generate forward slashes.

pkozlowski-opensource commented:
This PR was merged into the repository by commit bdbf21d04ba74a6f73469242076d6ce697c57edf.
